python/grass/jupyter/timeseries.py

```diff
@@ -0,0 +1,274 @@
+            elif self._etype == "stvds":
+                img.d_vect(map=name)
+            if self.overlay:
+                img.d_vect(map=self.overlay)
+            # Add legend if called
+            if self._legend:
+                info = gs.parse_command("t.info", input="precip_sum", flags="g")
+                min_min = info["min_min"]
+                max_max = info["max_max"]
+                img.d_legend(
+                    raster=name, range=f"{min_min}, {max_max}", **self._legend_kwargs
+                )
+
+    def time_slider(self, slider_width="60%"):
+        """
+        Create interactive timeline slider.
+
+        param str slider_width: width of datetime selection slider as a
+                                percentage (%) or pixels (px)
+        """
+        # Lazy Imports
+        import ipywidgets as widgets  # pylint: disable=import-outside-toplevel
+        from IPython.display import Image  # pylint: disable=import-outside-toplevel
+
+        # Datetime selection slider
+        slider = widgets.SelectionSlider(
+            options=self._dates,
+            value=self._dates[0],
+            description="Date/Time",
+            disabled=False,
+            continuous_update=True,
+            orientation="horizontal",
+            readout=True,
+            layout=widgets.Layout(width=slider_width),
+        )
+
+        # Display image associated with datetime
+        def view_image(date):
+            # Look up raster name for date
+            name = self._date_name_dict[date]
+            filename = os.path.join(self._tmpdir.name, f"{name}.png")
+            return Image(filename)
+
+        # Return interact widget with image and slider
+        widgets.interact(view_image, date=slider)
+
+    def animate(
+        self,
+        duration=500,
+        label=True,
+        font="DejaVuSans.ttf",
+        text_size=12,
+        text_color="gray",
+        filename=None,
+    ):
+        """
+        Creates a GIF animation of rendered layers.
+
+        Text color must be in a format accepted by PIL ImageColor module. For supported
+        formats, visit:
+        https://pillow.readthedocs.io/en/stable/reference/ImageColor.html#color-names
+
+        param int duration: time to display each frame; milliseconds
+        param bool label: include date/time stamp on each frame
+        param str font: font file
+        param int text_size: size of date/time text
+        param str text_color: color to use for the text.
+        param str filename: name of output GIF file
+        """
+        # Create a GIF from the PNG images
+        from PIL import Image  # pylint: disable=import-outside-toplevel
+        from PIL import ImageFont  # pylint: disable=import-outside-toplevel
+        from PIL import ImageDraw  # pylint: disable=import-outside-toplevel
+        from IPython.display import (  # pylint: disable=import-outside-toplevel
+            Image as ipyImage,
+        )
```

Comment:
You can `import IPython.display` and use IPython.display.Image to avoid issues with creating a good enough name.